DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs

```diff
@@ -577,13 +590,23 @@ private void GameLoadingLobby_GameLeft(object sender, EventArgs e)
             topBar.SwitchToSecondary();
             isInGameRoom = false;
             SetLogOutButtonText();
+
+            // keep the friends window up to date so it can disable the Invite option
+            pmWindow.inviteChannelName = string.Empty;
+            pmWindow.inviteGameName = string.Empty;
+            pmWindow.inviteChannelPassword = string.Empty;
         }
 
         private void GameLobby_GameLeft(object sender, EventArgs e)
         {
             topBar.SwitchToSecondary();
             isInGameRoom = false;
             SetLogOutButtonText();
+
+            // keep the friends window up to date so it can disable the Invite option
+            pmWindow.inviteChannelName = string.Empty;
+            pmWindow.inviteGameName = string.Empty;
+            pmWindow.inviteChannelPassword = string.Empty;
```

Comment:
This is duplicate code with lines 594 to 597, transfer them into a method. Preferably the method should be in `PrivateMessageWindow` because its fields / properties are the ones being modified.

Reply:
Added new SetInviteChannelInfo and ClearInviteChannelInfo methods to PrivateMessageWindow.